mm2src/lp_ordermatch.rs

```diff
@@ -4209,22 +4223,15 @@ pub async fn update_maker_order(ctx: &MmArc, req: MakerOrderUpdateReq) -> Result
         new_price
     ));
 
-    let mut my_maker_orders = ordermatch_ctx.my_maker_orders.lock().await;
-    match my_maker_orders.get_mut(&req.uuid) {
-        None => ERR!("Order with UUID: {} has been deleted", req.uuid),
-        Some(order) => {
-            if order.matches.len() != matches.len() || !order.matches.keys().all(|k| matches.contains_key(k)) {
-                return ERR!("Order {} is being matched now, can't update", req.uuid);
-            }
-
-            let new_change = HistoricalOrder::build(&update_msg, order);
-            order.apply_updated(&update_msg);
-            save_maker_order_on_update(ctx.clone(), order, new_change).await;
-            update_msg.with_new_max_volume((new_volume - reserved_amount).into());
-            maker_order_updated_p2p_notify(ctx.clone(), order.orderbook_topic(), update_msg).await;
-            Ok(order.clone())
-        },
+    let order_before_update = order.clone();
```

Comment:
As I can see, the `order` MutexGuard exists during all the balance checks and other IO, which may take a significant time. We should avoid holding the guards for such a long time unless we have a very reasoned requirement for such usage.